modules/ai/assets/js/editor/utils/editor-integration.js

```diff
@@ -30,17 +30,32 @@ export const getUiConfig = () => {
 	};
 };
 
+export const createScreenshot = async ( template ) => {
+	const screenshot = await takeScreenshot( template );
+
+	return {
+		screenshot,
+		template,
+	};
+};
+
 export const renderLayoutApp = ( options = {
 	at: null,
+	onClose: null,
+	onGenerate: null,
 	onInsert: null,
+	onRenderApp: null,
+	onSelect: null,
+	attachments: [],
 } ) => {
 	closePanel();
 
 	const previewContainer = createPreviewContainer( {
+		// Create the container at the "drag widget here" area position.
 		at: options.at,
 	} );
 
-	previewContainer.init();
+	options.onRenderApp?.( { previewContainer } );
```

Comment:
maybe put it after the `ReactDOM.render`?
or if it needs to be here, maybe change the name?
because `onRender` is usually after the render